src/js_object_accessor.hpp

```diff
@@ -421,33 +434,33 @@ struct Unbox<JSEngine, Timestamp> {
 
 template<typename JSEngine>
 struct Unbox<JSEngine, Obj> {
-    static Obj call(NativeAccessor<JSEngine> *ctx, typename JSEngine::Value const& value, realm::CreatePolicy policy, ObjKey current_row) {
+    static Obj call(NativeAccessor<JSEngine> *native_accessor, typename JSEngine::Value const& value, realm::CreatePolicy policy, ObjKey current_row) {
         using Value = js::Value<JSEngine>;
         using ValueType = typename JSEngine::Value;
 
-        auto object = Value::validated_to_object(ctx->m_ctx, value);
-        if (js::Object<JSEngine>::template is_instance<RealmObjectClass<JSEngine>>(ctx->m_ctx, object)) {
-            auto realm_object = get_internal<JSEngine, RealmObjectClass<JSEngine>>(ctx->m_ctx, object);
-            if (realm_object->realm() == ctx->m_realm) {
-                return realm_object->obj();
-            }
+        RealmLink<JSEngine> realm_link {native_accessor->m_ctx, value};
+        auto current_realm = native_accessor->m_realm;
 
-            bool updating = policy.copy && policy.update;
-            if (!updating && !policy.create) {
-                throw std::runtime_error("Realm object is from another Realm");
-            }
+        if(realm_link.belongs_to_realm(current_realm)){
+            return realm_link.get_realm_object();
+        }
+
+        if(realm_link.is_instance() && realm_link.is_read_only(policy)){
+            throw std::runtime_error("Realm object is from another Realm");
         }
```

Comment:
Why does the fact that a realm link is read only imply that it belongs to another Realm? Either the naming of the method is off or the error message is.